omod-2.4/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs2_4/TaskActionController2_4Test.java

```diff
@@ -0,0 +1,204 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.webservices.rest.web.v1_0.controller.openmrs2_4;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.openmrs.module.webservices.helper.TaskAction;
+import org.openmrs.module.webservices.rest.web.MockTaskServiceWrapper2_4;
+import org.openmrs.module.webservices.rest.web.api.RestService;
+import org.openmrs.module.webservices.rest.web.v1_0.controller.MainResourceControllerTest;
+import org.openmrs.module.webservices.rest.web.v1_0.resource.openmrs1_8.TaskActionResource1_8;
+import org.openmrs.module.webservices.rest.web.v1_0.resource.openmrs2_4.TaskDefinitionResource2_4;
+import org.openmrs.scheduler.Task;
+import org.openmrs.scheduler.TaskDefinition;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import java.util.Arrays;
+import java.util.Date;
+
+import static org.hamcrest.CoreMatchers.hasItem;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.core.IsNot.not;
+import static org.junit.Assert.assertEquals;
+
+public class TaskActionController2_4Test extends MainResourceControllerTest {
+
+	@Autowired
+	RestService restService;
+
+	static int count = 1;
+
```

Comment:
Can't we reuse some of the code that is already in `TaskActionController1_8Test`?

Reply:
I had trouble working with different contexts but I will look into that.